src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -79,4 +84,59 @@ public void pollEvents(final LocalDateTime currentTime) {
             log.warn("Skipping Event Retrieval as start after end, start = {}, end = {}", startTime, endTime);
         }
     }
+
+    /**
+     * For testing run a poll for the immediately ended minute, repeat 1 min later and compare:
+     * <ol>
+     *  <li>run for the latest minute (POLL_NAME_TEST.time -> now - 1 second) remembering time range and count, A
+     *  <li>run for [PREVIOUS_POLL_NAME_TEST -> POLL_NAME_TEST] time range remembering time range and count, B
+     *  <li>compare count B with POLL_NAME_TEST.count - should be the same
+     *  <li>store count A in POLL_NAME_TEST.count
+     *  <li>move POLL_NAME_TEST.time to PREVIOUS_POLL_NAME_TEST.time
+     *  <li>store A end time in POLL_NAME_TEST.time
+     * </ol>
+     */
+    public void runTestPolls(LocalDateTime now) {
+        final LocalDateTime endTimeA = now.minus(1, SECONDS);
+
+        repository.findById(POLL_NAME_TEST).ifPresentOrElse(
+                test -> {
+                    val previousTest = repository.findById(PREVIOUS_POLL_NAME_TEST).orElse(PollAudit.builder()
```

Comment:
when will `findById(PREVIOUS_POLL_NAME_TEST)` be not found?  Isn't that what the `OrElse` on line 102 do anyway?

Reply:
I create both POLL_NAME_TEST and PREVIOUS_POLL_NAME_TEST together so if the first exists then so does the 2nd

Reply:
oh i see what you mean ! ok